### PR TITLE
[c10d][Logging] Remove args and kwargs from c10d logging

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140169

This PR is trying to reland https://github.com/pytorch/pytorch/pull/139804

We now don't want to log args and kwargs directly because if they contain tensor or tensor subclass it would take lots of time in conversion to string or even not supported.

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by wz337 — APPROVED** [review]
LGTM!

**Comment by fduwjj** [comment]
@ZainRizvi why are you saying that, I run the test locally and it pass for me. And I added even the label of `ci-no-td`. What else should we do?

File: test/distributed/test_c10d_logger.py [hunk 1]
```diff
@@ -118,16 +118,14 @@ def test_exception_logger(self) -> None:
                 re.search("({.+})", captured.output[0]).group(0).replace("'", '"')
             )
 
-            self.assertEqual(len(error_msg_dict), 10)
+            self.assertEqual(len(error_msg_dict), 9)
```
Comment (kwen2501) [inline test/distributed/test_c10d_logger.py:121]: nit: remove this check entirely to reduce our maintenance overhead?

File: test/distributed/test_c10d_logger.py [hunk 2]
```diff
@@ -162,16 +160,14 @@ def test_time_logger(self) -> None:
             msg_dict = json.loads(
                 re.search("({.+})", captured.output[0]).group(0).replace("'", '"')
             )
-            self.assertEqual(len(msg_dict), 10)
+            self.assertEqual(len(msg_dict), 9)
```
Comment (kwen2501) [inline test/distributed/test_c10d_logger.py:163]: Same here.

**Review by kwen2501 — APPROVED** [review]
Big fix!

File: test/distributed/test_c10d_logger.py (hunk 1 above)
Comment (fduwjj) [reply test/distributed/test_c10d_logger.py:121]: why is that? This is part of a test. Having a proper test is not a maintenance overhead.

**Pull request closed without merging**